dali/pipeline/data/tensor_vector.h

```diff
@@ -130,358 +99,85 @@ class TensorVector {
     return tensors_.size();
   }
 
-  size_t nbytes() const noexcept {
-    if (state_ == State::contiguous) {
-      return tl_->nbytes();
-    }
-    // else
-    size_t total_nbytes = 0;
-    for (const auto &t : tensors_) {
-      total_nbytes += t->nbytes();
-    }
-    return total_nbytes;
-  }
+  size_t nbytes() const noexcept;
 
-  size_t capacity() const noexcept {
-    if (state_ == State::contiguous) {
-      return tl_->capacity();
-    }
-    // else
-    size_t total_capacity = 0;
-    for (const auto &t : tensors_) {
-      total_capacity += t->capacity();
-    }
-    return total_capacity;
-  }
+  size_t capacity() const noexcept;
 
-  TensorListShape<> shape() const {
-    if (state_ == State::contiguous) {
-      return tl_->shape();
-    }
-    if (tensors_.empty()) {
-      return {};
-    }
-    TensorListShape<> result(tensors_.size(), tensors_[0]->ndim());
-    for (size_t i = 0; i < tensors_.size(); i++) {
-      result.set_tensor_shape(i, tensors_[i]->shape());
-    }
-    return result;
-  }
+  TensorListShape<> shape() const;
 
-  DLL_PUBLIC inline void Resize(const TensorListShape<> &new_shape) {
-    return Resize(new_shape, type());
-  }
+  DLL_PUBLIC void Resize(const TensorListShape<> &new_shape);
 
-  DLL_PUBLIC inline void Resize(const TensorListShape<> &new_shape, const TypeInfo &new_type) {
-    // N.B. There probably is nothing wrong with adjusting batchsize for give TensorVector
-    // (sparse tensor list), but the semantics of what type and pinned status should
-    // the new elements have or having some of them allocated and the new ones not
-    // complicates the logic even further so we disallow it
-    DALI_ENFORCE(tensors_.empty() || static_cast<int>(tensors_.size()) == new_shape.size(),
-                 "Changing the batch size is prohibited. It should be set once.");
-    if (tensors_.empty()) {
-      allocate_tensors(new_shape.size());
-    }
-    if (state_ == State::contiguous) {
-      tl_->Resize(new_shape, new_type);
-      UpdateViews();
-      return;
-    }
-    for (int i = 0; i < new_shape.size(); i++) {
-      tensors_[i]->Resize(new_shape[i], new_type);
-    }
-  }
+  DLL_PUBLIC void Resize(const TensorListShape<> &new_shape, const TypeInfo &new_type);
 
-  inline void set_type(const TypeInfo &new_type) {
-    type_ = new_type;
-    tl_->set_type(new_type);
-    for (auto t : tensors_) {
-      t->set_type(new_type);
-    }
-    if (state_ == State::contiguous) {
-      UpdateViews();
-    }
-  }
+  void set_type(const TypeInfo &new_type);
 
-  inline const TypeInfo &type() const {
-    if (state_ == State::contiguous) {
-      return tl_->type();
-    }
-    if (tensors_.size() > 0) {
-      return tensors_[0]->type();
-    }
-    return type_;
-  }
+  const TypeInfo &type() const;
```

Comment:
Please leave such simple getters as inline functions.

Reply:
I don't know, if this getter is yet a simple one